lib/battle/CBattleInfoEssentials.cpp

```diff
@@ -47,10 +47,7 @@ std::vector<std::shared_ptr<const CObstacleInstance>> CBattleInfoEssentials::bat
 	else
 	{
 		if(!!player && *perspective != battleGetMySide())
-		{
-			logGlobal->error("Unauthorized obstacles access attempt!");
-			return ret;
-		}
+			logGlobal->warn("Unauthorized obstacles access attempt, assuming massive spell");

```

Comment:
In which scenarios could this happen?

Reply:
@IvanSavenko when we ask if we can use expert dispel and there is a landmines placed by enemy and no other spell. Before this fix we will get "this spell will affects nothing" message. With this change mines are dispelled correctly.